File: Cc.Core/Models/TabDistrict.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class TabDistrict
    {
        public TabDistrict()
        {
            Customer = new HashSet<Customer>();
            TabCity = new HashSet<TabCity>();
        }

        public long Id { get; set; }
        public DateTime InsertDate { get; set; }
        public string? InsertUser { get; set; }
        public DateTime UpdateDate { get; set; }
        public string? UpdateUser { get; set; }
        public long TabRegionId { get; set; }
        public string Code { get; set; } = null!;
        public string? IstatCode { get; set; }
        public string? Description { get; set; }

        public virtual TabRegion TabRegion { get; set; } = null!;
        public virtual ICollection<Customer> Customer { get; set; }
        public virtual ICollection<TabCity> TabCity { get; set; }
    }
}

File: Cc.Ui/src/models/tabDistrict.ts
import { Entity, resource, field, association } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';
import { TabRegion } from './tabRegion';

@resource('tabDistrict')
export class TabDistrict extends Entity {

    @field()
    id: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;
    @field()
    tabRegionId: number;
    
    @field()
    code: string;
    @field()
    istatCode: string;
    @field()
    description: string;
    
    @association(TabRegion, 'tabRegionId')
    tabRegion: TabRegion;

    guid: string;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): TabDistrict {

        let item = new TabDistrict();
        return item;
    }
}

@resource('tabDistrict')
export class TabDistrictSearchModel extends Entity implements Pagination {
    
    @field()
    tabRegionId: number;
    
    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}